flare/kernels/mc_sephyps.py

```diff
@@ -65,161 +103,318 @@
+
+
+def two_three_many_mc_force_en(env1, env2, d1, cutoff_2b, cutoff_3b, cutoff_mb,
                                nspec, spec_mask,
                                nbond, bond_mask, ntriplet, triplet_mask,
+                               ncut3b, cut3b_mask,
+                               nmb, mb_mask,
                                sig2, ls2, sig3, ls3, sigm, lsm,
                                cutoff_func=cf.quadratic_cutoff):
-    r_cut_2 = cutoffs[0]
-    r_cut_3 = cutoffs[1]
-    r_cut_m = cutoffs[2]
+    """2+3+manybody multi-element kernel between a force component and a local
+    energy.
+
+    Args:
+        env1 (AtomicEnvironment): First local environment.
+        env2 (AtomicEnvironment): Second local environment.
+        d1 (int): Force component of the first environment.
+        d2 (int): Force component of the second environment.
+        cutoff_2b (float, np.ndarray): cutoff(s) for two-body interaction
+        cutoff_3b (float, np.ndarray): cutoff(s) for three-body interaction
+        cutoff_mb (float, np.ndarray): cutoff(s) for coordination-based manybody interaction
+        nspec (int): number of different species groups
+        spec_mask (np.ndarray): 118-long integer array that determines specie group
+        nbond (int): number of different hyperparameter sets to associate with two-body pairings
+        bond_mask (np.ndarray): nspec^2 long integer array
+        ntriplet (int): number of different hyperparameter sets to associate with 3-body pairings
+        triplet_mask (np.ndarray): nspec^3 long integer array
+        ncut3b (int): number of different 3-body cutoff sets to associate with 3-body pairings
+        cut3b_mask (np.ndarray): nspec^2 long integer array
+        nmb (int): number of different hyperparameter sets to associate with manybody pairings
+        mb_mask (np.ndarray): nspec^2 long integer array
+        sig2 (np.ndarray): signal variances associates with two-body term
+        ls2 (np.ndarray): length scales associates with two-body term
+        sig3 (np.ndarray): signal variances associates with three-body term
+        ls3 (np.ndarray): length scales associates with three-body term
+        sigm (np.ndarray): signal variances associates with manybody term
+        lsm (np.ndarray): length scales associates with manybody term
+        cutoff_func (Callable): Cutoff function of the kernel.
+
+    Return:
+        float: Value of the 2+3+many-body force/energy kernel.
+    """
 
     two_term = \
         two_body_mc_force_en_jit(env1.bond_array_2, env1.ctype, env1.etypes,
                                  env2.bond_array_2, env2.ctype, env2.etypes,
-                                 d1, sig2, ls2, r_cut_2, cutoff_func,
+                                 d1, sig2, ls2, cutoff_2b, cutoff_func,
                                  nspec, spec_mask,
                                  bond_mask) / 2
 
+    if (ncut3b == 0):
+        tbmcj = three_body_mc_force_en_jit
```

Comment:
I think we can afford to make this more verbose :) `tb_mc_jit_func?

Reply:
hahaha . Sure. because the variable is used immediately in the next line. So I just make it very short...